x-pack/plugins/cases/public/components/attachments/attachments_table.tsx

```diff
@@ -0,0 +1,189 @@
+  };
+
+  const columns: Array<EuiBasicTableColumn<FileJSON>> = [
+    {
+      name: 'Name',
+      'data-test-subj': 'attachments-table-filename',
+      render: (attachment: FileJSON) => (
+        <EuiLink onClick={() => showModal(attachment)}>{attachment.name}</EuiLink>
+      ),
+      width: '60%',
+    },
+    {
+      field: 'mimeType',
+      'data-test-subj': 'attachments-table-filetype',
+      name: 'Type',
+    },
+    {
+      field: 'created',
+      name: 'Date Added',
+      'data-test-subj': 'attachments-table-date-added',
+      dataType: 'date',
+    },
+    {
+      name: 'Actions',
+      width: '120px',
+      actions: [
+        {
+          name: 'Download',
+          isPrimary: true,
+          description: 'Download this file',
+          render: (attachment: FileJSON) => {
+            return (
+              <EuiButtonIcon
+                iconType={'download'}
+                aria-label={'download'}
+                href={filesClient.getDownloadHref({
+                  fileKind: CASES_FILE_KINDS[APP_ID].id,
+                  id: attachment.id,
+                })}
+              />
+            );
+          },
+          'data-test-subj': 'attachments-table-action-download',
+        },
+        {
+          name: 'Delete',
+          isPrimary: true,
+          description: 'Delete this file',
+          color: 'danger',
+          icon: 'trash',
+          type: 'icon',
+          onClick: () => {},
+          'data-test-subj': 'attachments-table-action-delete',
+        },
+      ],
+    },
+  ];
+
+  const resultsCount = useMemo(
+    () => (
+      <>
+        <strong>
+          {pagination.pageSize * pagination.pageIndex + 1}
+          {'-'}
+          {pagination.pageSize * pagination.pageIndex + pagination.pageSize}
+        </strong>{' '}
+        {'of'} <strong>{pagination.totalItemCount}</strong>
+      </>
+    ),
+    [pagination.pageIndex, pagination.pageSize, pagination.totalItemCount]
+  );
+
+  return isLoading ? (
+    <EuiLoadingContent data-test-subj="attachments-table-loading" lines={10} />
+  ) : (
+    <>
+      {pagination.totalItemCount > 0 && (
+        <>
+          <EuiSpacer size="xl" />
+          <EuiText size="xs" data-test-subj="attachments-table-results-count">
+            {'Showing'} {resultsCount}
+          </EuiText>
+        </>
+      )}
+      <EuiSpacer size="s" />
+      <EuiBasicTable
+        tableCaption="Attachments Table"
+        items={items}
+        columns={columns}
+        pagination={pagination}
+        onChange={onChange}
+        data-test-subj="attachments-table"
+        noItemsMessage={
+          <EuiEmptyPrompt
+            title={<h3>{'No attachments available'}</h3>}
+            data-test-subj="attachments-table-empty"
+            titleSize="xs"
+            actions={
+              <EuiButton
+                size="s"
+                iconType="plusInCircle"
+                data-test-subj="case-detail-attachments-table-upload-file"
```

Comment:
Don't forget to add an even handler `onClick` for the button.

Reply:
There is already an `href` is that necessary?